### ISSUE TITLE
Star recent search

### ISSUE THREAD

**Issue by tadashi-aikawa** [tadashi-aikawa/obsidian-another-quick-switcher#37]
| Priority | Recent Search | Star Recent Search |
| -------- | ------------- | ------------------ |
| 1        | Last opened   | **Starred**        |
| 2        | Last modified | Last opened        |
| 3        | -             | Last modified      |
| 4        | -             | Filename matched   |

**Comment by chrisgrieser** [comment]
I actually have already written a "starred items only" switcher as a templater script. Also sorts item's by recency. Feel free to use (parts) of the code if you want 🙂 


```js
<%*
// array contains 10 last notes, with lower index being more recent
const recentFiles = app.workspace.lastOpenFiles;

// get starred files
const starredFiles = app.internalPlugins.plugins.starred.instance.items
	.filter (item => item.type === "file")
	.filter (item => tp.file.exists(item.path)) 
	.sort((a, b) => {
		const aIsRecent = recentFiles.includes(a.path);
		const bIsRecent = recentFiles.includes(b.path);
		if (!aIsRecent && !bIsRecent) return 0;
		if (aIsRecent && !bIsRecent) return -1;
		if (!aIsRecent && bIsRecent) return 1;

		return recentFiles.indexOf(a.path) - recentFiles.indexOf(b.path)
	});

starredNames = starredFiles.map (item => item.title);
starredPaths = starredFiles.map (item => item.path);

// Suggester https://silentvoid13.github.io/Templater/internal-functions/internal-modules/system-module.html
const selectedNote = await tp.system.suggester(starredNames, starredPaths, true, "☆"); // placeholder ("☆") used by css snippet for targetting

// Save file also triggers linting and refresh breadcrumbs
setTimeout(() => {
	const tFileToOpen = tp.file.find_tfile(selectedNote);
	const activePane = this.app.workspace.activeLeaf;
	const currentView = activePane.getViewState();
	activePane.openFile(tFileToOpen, currentView);
}, 250);

// open the selected note
const tFileToOpen = tp.file.find_tfile(selectedNote); // get tfile for the path
const activePane = this.app.workspace.activeLeaf;
const currentView = activePane.getViewState();
activePane.openFile(tFileToOpen, currentView);

return;
%>
```

**Comment by tadashi-aikawa** [comment]
Thanks @chrisgrieser! 😄  I will refer to it 👍

**Comment by tadashi-aikawa** [comment]
Released [v4.13.0](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.13.0) 🚀
